### PR TITLE
Add wifi.joinWithAgent

### PR DESCRIPTION
**Pull request by pcarranzav**
Allows joining a secured network without accessing connman files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
   },
   "license": "MIT",
   "dependencies": {
-    "connman-api": "git+https://github.com/Doodle3D/connman-api.git",
+    "connman-api": "git+https://github.com/pcarranzav/connman-api.git#5bae8c4d15066ac36a57069c2003901e5dee398d",
```
Comment (pcarranzav) [inline package.json]: This should be changed to the corresponding connman-api version once the changes from [#7](https://github.com/Doodle3D/connman-api/pull/7) are published to the npm registry.

Comment (peteruithoven) [inline package.json]: Id'd rather first publish on npm.

**Comment by pcarranzav** [comment]
Added a commit that throws out the config files hack... now it's simpler I guess :)

File: package.json (hunk 1 above)
Comment (peteruithoven) [inline package.json]: I published a new version of connman-api on npm.

**Comment by peteruithoven** [comment]
Thanks for the cooperation!

**Comment by pcarranzav** [comment]
You're welcome! Just opened #15 because I didn't realize we still need the config files to forget networks...

**Pull request merged**